ipfs/embedded.go

```diff
@@ -25,70 +22,62 @@ import (
 
 // Embedded is the provider that embeds IPFS node within the same process.
 // It also returns closable for graceful node shutdown.
-func Embedded(init bool, cfg *Config, logger log.Logger) APIProvider {
-	return func() (coreiface.APIDagService, io.Closer, error) {
-		path := cfg.Path()
-		defer os.Setenv(ipfscfg.EnvDir, path)
+func Embedded(init bool, cfg *Config, logger log.Logger) (APIProvider, error) {
+	path := cfg.Path()
+	defer os.Setenv(ipfscfg.EnvDir, path)
```

Comment:
Let's keep IPFS providing lazy. Any reason to change that?

Reply:
That's a great point, it would be nice to not init the ipfs node until API provider is called. The reason it was changed was so that we could use an interface instead of a typed function. Although, I'm not sure of the benefit given that even in the case that it is lazy, it is called immediately after being passed to the `node.NewNode` func